controllers/helmchart_controller.go

```diff
@@ -373,37 +375,71 @@ func (r *HelmChartReconciler) reconcileFromHelmRepository(ctx context.Context,
 		return sourcev1.HelmChartNotReady(chart, sourcev1.ChartPullFailedReason, err.Error()), err
 	}
 
+	var valuesFiles []string
+	// Prepend a list of valuesFiles with the deprecated ValuesFile.
+	if chart.Spec.ValuesFile != "" {
+		valuesFiles = append(valuesFiles, chart.Spec.ValuesFile)
+	}
+	valuesFiles = append(valuesFiles, chart.Spec.ValuesFiles...)
```

Comment:
```suggestion
	valuesFiles := chart.Spec.ValuesFiles
	// Append the deprecated ValuesFile to the list
	valuesFile = append([]string{chart.Spec.ValuesFile}, valuesFiles...)
```

would make the deprecation helpers easier to remove in an upcoming version.

We may also want to simply move this logic to a `GetValuesFiles() []string` method on the `HelmRelease` object so the duplication of this here and on `L543-548` is factored out.

Reply:
Done.